.github/workflows/security-scan.yml

```diff
@@ -0,0 +1,92 @@
+name: Security Scan
+
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+    branches:
+      - main
+      - release/**
+
+# cancel existing runs of the same workflow on the same ref
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  conditional-skip:
+    runs-on: ubuntu-latest
+    name: Get files changed and conditionally skip CI
+    outputs:
+      skip-ci: ${{ steps.read-files.outputs.skip-ci }}
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          fetch-depth: 0
+      - name: Get changed files
+        id: read-files
+        run: ./.github/scripts/filter_changed_files_go_test.sh
+
+  setup:
+    needs: [conditional-skip]
+    name: Setup
+    if: needs.conditional-skip.outputs.skip-ci != 'true'
+    runs-on: ubuntu-latest
+    outputs:
+      compute-small: ${{ steps.setup-outputs.outputs.compute-small }}
+      compute-medium: ${{ steps.setup-outputs.outputs.compute-medium }}
+      compute-large: ${{ steps.setup-outputs.outputs.compute-large }}
+      compute-xl: ${{ steps.setup-outputs.outputs.compute-xl }}
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - id: setup-outputs
+        name: Setup outputs
+        run: ./.github/scripts/get_runner_classes.sh
+
+  scan:
+    needs: [setup]
+    runs-on: ${{ fromJSON(needs.setup.outputs.compute-large) }}
+    # The first check ensures this doesn't run on community-contributed PRs, who
+    # won't have the permissions to run this job.
+    if: ${{ (github.repository != 'hashicorp/consul' || (github.event.pull_request.head.repo.full_name == github.event.pull_request.base.repo.full_name))
+      && (github.actor != 'dependabot[bot]') && (github.actor != 'hc-github-team-consul-core') }}
+
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Set up Go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          cache: true
+          go-version: 1.20.12  #TODO move CI build config and this to .go-version or .go-mod
+
+      - name: Clone Security Scanner repo
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          repository: hashicorp/security-scanner
+          token: ${{ secrets.ELEVATED_GITHUB_TOKEN }}  #TODO replace w/ security-provided secret
```

Comment:
I've noticed other repos (e.g. [Vault](https://github.com/hashicorp/vault/blob/main/.github/workflows/security-scan.yml#L42)) use a `HASHIBOT_PRODSEC_GITHUB_TOKEN` secret for this checkout. If that's something that we can add now, I'll update this before merging.

Reply:
We need a token that can checkout the private repository. Hopefully in the near future, we can make this a public repository and avoid the extra syntax/burden here.

So, `HASHIBOT_PRODSEC_GITHUB_TOKEN` or anything else that can view/clone the `hashicorp/security-scanner` repository will work.

Reply:
Sounds good, and the one here certainly passes that bar. We can continue to use it, but I'd prefer a least-privilege token if that's an option. Does Security typically generate a `HASHIBOT_PRODSEC_GITHUB_TOKEN` for teams?

Reply:
IIRC, this secret should _already_ be available for this repository's workflows to use. If that's not the case for some reason, we can get that sorted out in the near future 👍

Reply:
Thanks! I think we'll need one added; I checked the secrets and couldn't find it available, and this test seems to confirm we're behind the times: https://github.com/hashicorp/consul/actions/runs/7254694293/job/19763912516#step:4:15

Reply:
We'll look into getting that sorted out to unblock this PR! 😄

Reply:
👋  @zalimeni, `HASHIBOT_PRODSEC_GITHUB_TOKEN`has been added to the repo secrets